File: tests/generic/gen_cor_073/Input/event_listener.js
const Web3 = require('web3');
const commander = require('commander');
const fs = require('fs');

require('console-stamp')(console, 'HH:MM:ss');

function task(contract, from) {
  setTimeout(function() {
    contract.getPastEvents('Stored', { fromBlock: from, toBlock: 'latest'})
    .then(function(events) {
        if (events.length) {
            for (var i = 0, len = events.length; i < len; i+=1) {
                console.log(events[i]);
                from = events[i].blockNumber
            }
        }
        task(contract, from+1)
    });
  }, 2000);
}

commander
  .version('1.0.0', '-v, --version')
  .usage('[OPTIONS]...')
  .option('-u, --url <url>', 'Connection URL')
  .option('-a, --address <value>', 'Address of the contract')
  .option('-b, --abi <value>', 'Abi of the contract')
  .option('-p, --pk <value>', 'Private key of account to poll')
  .parse(process.argv);

const options = commander.opts();
console.log('URL:', `${options.url}`);
console.log('ADR:', `${options.address}`);
console.log('ABI:', `${options.abi}`);
console.log('PK:', `${options.pk}`);

const json = fs.readFileSync(`${options.abi}`);
const abi = JSON.parse(json);

const web3 = new Web3(`${options.url}`);
const contract = new web3.eth.Contract(abi, `${options.address}`)

console.log('Starting to run the event loop')
task(contract, 0);

File: tests/generic/gen_cor_072/Input/event_listener.py
from web3 import Web3
import logging
import requests, time
import argparse, json, sys
from eth_account.messages import encode_defunct

logging.basicConfig(format='%(asctime)s %(message)s', stream=sys.stdout, level=logging.INFO)


def generate_viewing_key(web3, url, private_key):
    logging.info('Generating viewing key for %s' % private_key)

    account = web3.eth.account.privateKeyToAccount(private_key)

    headers = {'Accept': 'application/json', 'Content-Type': 'application/json'}
    data = {"address": account.address}
    response = requests.post('%s/generateviewingkey/' % url, data=json.dumps(data), headers=headers)

    signed_msg = web3.eth.account.sign_message(encode_defunct(text='vk' + response.text), private_key=private_key)
    data = {"signature": signed_msg.signature.hex(), "address": account.address}
    requests.post('%s/submitviewingkey/' % url, data=json.dumps(data), headers=headers)


if __name__ == "__main__":
    parser = argparse.ArgumentParser(prog='event_listener')
    parser.add_argument('-u', '--url', help='Connection URL')
    parser.add_argument('-a', '--address', help='Address of the contract')
    parser.add_argument('-b', '--abi', help='Abi of the contract')
    parser.add_argument('-p', '--pk', help='Private key of account to poll')
    parser.add_argument('--obscuro', action='store_true', help='True if running against obscuro')
    args = parser.parse_args()

    logging.info('URL: %s' % args.url)
    logging.info('ADR: %s' % args.address)
    logging.info('ABI: %s' % args.abi)

    web3 = Web3(Web3.HTTPProvider(args.url))
    if args.obscuro: generate_viewing_key(web3, args.url, args.pk)
    with open(args.abi) as f:
        contract = web3.eth.contract(address=args.address, abi=json.load(f))

    logging.info('Starting to run the event loop')
    event_filter = contract.events.Stored.createFilter(fromBlock='latest')
    while True:
        for event in event_filter.get_new_entries():
            logging.info('Stored value = %s' % event['args']['value'])
        time.sleep(2)